src/main/java/org/folio/service/FundsDistributionService.java

```diff
@@ -29,54 +30,64 @@ public class FundsDistributionService {
+        .map(fundDistribution -> getDistributionAmount(fundDistribution, expectedTotal, poLineCurrency, conversion))
+        .reduce((money, money2) -> Money.from(MonetaryFunctions.sum(money, money2)))
+        .orElseGet(() -> Money.zero(poLineCurrency));
 
       MonetaryAmount remainder = expectedTotal.abs()
-          .subtract(calculatedTotal.abs());
+        .subtract(calculatedTotal.abs());
       int remainderSignum = remainder.signum();
       MonetaryAmount smallestUnit = getSmallestUnit(expectedTotal, remainderSignum);
 
-      for (ListIterator<EncumbranceRelationsHolder> iterator = getIterator(encumbranceRelationsHolders, remainderSignum); isIteratorHasNext(iterator, remainderSignum);) {
+      for (var iterator = getIterator(encumbranceRelationsHolders, remainderSignum); isIteratorHasNext(iterator, remainderSignum); ) {
 
-        final EncumbranceRelationsHolder holder = iteratorNext(iterator, remainderSignum);
-          CurrencyUnit fyCurrency = Monetary.getCurrency(holder.getCurrency());
-          MonetaryAmount initialAmount = getDistributionAmount(holder.getFundDistribution(), expectedTotal, poLineCurrency, conversion);
+        EncumbranceRelationsHolder holder = iteratorNext(iterator, remainderSignum);
+        CurrencyUnit fyCurrency = Monetary.getCurrency(holder.getCurrency());
+        MonetaryAmount initialAmount = getDistributionAmount(holder.getFundDistribution(), expectedTotal, poLineCurrency, conversion);
 
-          if (FundDistribution.DistributionType.PERCENTAGE.equals(holder.getFundDistribution().getDistributionType()) && !remainder.isZero()) {
-            initialAmount = initialAmount.add(smallestUnit);
-            remainder = remainder.abs().subtract(smallestUnit.abs()).multiply(remainderSignum);
-          }
+        if (holder.getFundDistribution().getDistributionType() == FundDistribution.DistributionType.PERCENTAGE && !remainder.isZero()) {
+          initialAmount = initialAmount.add(smallestUnit);
+          remainder = remainder.abs().subtract(smallestUnit.abs()).multiply(remainderSignum);
+        }
 
-          MonetaryAmount expended = Optional.of(holder).map(EncumbranceRelationsHolder::getNewEncumbrance).map(Transaction::getEncumbrance).map(Encumbrance::getAmountExpended).map(aDouble -> Money.of(aDouble, fyCurrency))
-              .orElse(Money.zero(fyCurrency));
-          MonetaryAmount awaitingPayment = Optional.of(holder).map(EncumbranceRelationsHolder::getNewEncumbrance).map(Transaction::getEncumbrance).map(Encumbrance::getAmountAwaitingPayment)
-              .map(aDouble -> Money.of(aDouble, fyCurrency)).orElse(Money.zero(fyCurrency));
-          MonetaryAmount amount = calculateEncumbranceEffectiveAmount(initialAmount, expended, awaitingPayment, fyCurrency);
+        MonetaryAmount expended = getOrZero(holder, fyCurrency, FinanceUtils::getExpended);
+        MonetaryAmount awaitingPayment = getOrZero(holder, fyCurrency, Encumbrance::getAmountAwaitingPayment);
```

Comment:
getOrZero. confusing name, can be name getNonNegativeAmount, extractAmount ... smth other name is more self-explanatory.

Reply:
renamed